docs/develop/gettingstarted.rst

```diff
@@ -0,0 +1,300 @@
+   $ git clone https://github.com/<your-user-name>/micropython.git
+
+Depending on what you cloned, `configure the remote 
+repositories <https://git-scm.com/book/en/v2/Git-Basics-Working-with-Remotes>`_ to be able to
+collaborate on the MicroPython project.
+
+Configure upstream if you cloned your fork.
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+
+It is common to configure ``upstream`` and ``origin`` on a forked repository
+to assist with sharing code changes. You can maintain your own mapping but
+it is recommended that ``origin`` maps to your fork and ``upstream`` to the main
+MicroPython repository:
+
+.. code-block:: console
+
+   $ cd micropython
+   $ git remote add upstream https://github.com/micropython/micropython.git
+   $ git remote set-url origin https://github.com/<your-user-name>/micropython.git
+
+After the above configuration, your setup should be similar to this:
+
+.. code-block:: console
+   
+   $ git remote -v
+   origin	https://github.com/<your-user-name>/micropython.git (fetch)
+   origin	https://github.com/<your-user-name>/micropython.git (push)
+   upstream	https://github.com/micropython/micropython.git (fetch)
+   upstream	https://github.com/micropython/micropython.git (push)
+
+You should now have a copy of the source code. By default you are pointing
+to the master branch. To prepare for further development, it is recommended
+to work on a development branch.
+
+.. code-block:: console
+
+    $ git checkout -b dev-branch
+
+You can give it any name. You will have to compile MicroPython whenever you change 
+to a different branch.
+
+Compile and build the Code
+--------------------------
+
+For development, you should compile MicroPython for a specific platform which corresponds
+to a port. Start by installing the required dependencies. Then build the MicroPython 
+cross-compiler before you can successfully compile and build.
+
+Required dependencies
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Before anything, make sure you have a working environment for ``gcc`` and ``GNU make``.
+Ubuntu 20.04 has been used for the example below but other unixes ought to work with
+little modification:
+
+.. code-block:: console
+
+   $ gcc --version
+   gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0
+   Copyright (C) 2019 Free Software Foundation, Inc.
+   This is free software; see the source for copying conditions.  There is NO
+   warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+
+Install the required dependencies for linux:
+
+.. code-block:: console
+
+   $ sudo apt-get install build-essential libreadline-dev libffi-dev git \\
+   pkg-config gcc-arm-none-eabi libnewlib-arm-none-eabi
+
+For the stm32 port, the ARM cross-compiler is required:
+
+.. code-block:: console
+
+   $ sudo apt-get install arm-none-eabi-gcc arm-none-eabi-binutils arm-none-eabi-newlib
+
+Python 3 is also recommended:
```

Comment:
Python is a hard requirement to be able to build

Reply:
Required for recommended is better I guess?

Reply:
Indeed.
(technically speaking the statement 'Python 3 is required' is not completely correct because for the build itself Python 2 works as well, as far as I know. However for this document that's irrelevant)

Reply:
If Python 2 works then, leaving it as recommended sgtm, no?

Reply:
Sorry should have been more clear - 'build itself' is key here: the tests for instance do require Python 3, and there are some scripts in e.g. the tools/ directory which do so as well. tldr; for the purpose of this document Python3 should be listed like any other required dependency

Reply:
I made some edits based on @jimmo 's suggestions